src/flb_processor.c

```diff
@@ -264,15 +264,30 @@ struct flb_processor_unit *flb_processor_unit_create(struct flb_processor *proc,
     return pu;
 }
 
-int flb_processor_unit_set_property(struct flb_processor_unit *pu, const char *k, const char *v)
+int flb_processor_unit_set_property(struct flb_processor_unit *pu, const char *k, struct cfl_variant *v)
 {
+    struct cfl_variant *val;
+    int i;
+    int ret;
+
     if (pu->unit_type == FLB_PROCESSOR_UNIT_FILTER) {
-        return flb_filter_set_property(pu->ctx, k, v);
+        if (v->type == CFL_VARIANT_STRING) {
+            return flb_filter_set_property(pu->ctx, k, v->data.as_string);
+        } else if (v->type == CFL_VARIANT_ARRAY) {
```

Comment:
ditto.

Reply:
I went through and updated all instances of `if` to comply in the files `src/flb_processor.c` and `src/config_format/flb_cf_yaml.c`.